### PR TITLE
Fix: show latest version if it is an upload in activity feed

### PR DESCRIPTION
**Pull request by DanDeMicco**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @DanDeMicco has signed the CLA. Thanks for the pull request!

**Comment by boxcla** [comment]
# Webpack Bundle Size Github Bot

| Bundle        | Diff | Percentage Diff | Before | After |
| ------------- |:----:|:---------------:|:------:|:-----:|
| Total | 0 B | 0.00% | 0 B | 0 B |

Compared against previously merged PR #473

**Powered by [webpack-bundle-size-github-bot](https://github.com/wenboyu2/webpack-bundle-size-github-bot/)**

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 1]
```diff
@@ -424,30 +424,48 @@ class ActivityFeed extends React.Component<Props, State> {
...
-        const lastVersion = versions.total_count ? versions.entries[versions.total_count - 1] : {};
-        if (restored_from && lastVersion.action !== VERSION_RESTORE_ACTION) {
-            const restoredVersion = versions.entries.find((version) => version.id === restored_from.id);
+        if (file_version) {
+            if (restored_from) {
+                const entriesWithRestore: Array<BoxItemVersion> = versions.entries.map(
+                    (version: BoxItemVersion): BoxItemVersion => {
+                        if (version.id === restored_from.id) {
+                            return {
+                                ...version,
```
Comment (priyajeet) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: Thats a lot of arrowness

**Comment by DanDeMicco** [comment]
Pushed a new patchset

**Comment by pramodsum** [comment]
Make sure to check that the feed shows an empty state when you have a freshly uploaded file w/o comments/tasks and then displays the active state when you add a comment/task to the file

**Comment by DanDeMicco** [comment]
@pramodsum LGTM

File: src/api/__tests__/Versions-test.js [hunk 2]
```diff
@@ -43,14 +43,12 @@ describe('api/Versions', () => {
                 total_count: 2,
                 entries: [
                     {
-                        ...deleteVersion,
-                        action: 'delete',
-                        version_number: 1
```
Comment (pramodsum) [inline src/api/__tests__/Versions-test.js]: Since version_number now gets returned w/ the API response for versions, that field should probably remain in the tests

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 3]
```diff
@@ -424,30 +424,45 @@ class ActivityFeed extends React.Component<Props, State> {
...
-            if (restoredVersion) {
-                versions.entries.push({
-                    ...restoredVersion,
-                    // $FlowFixMe
-                    id: file_version.id,
-                    created_at: modified_at,
-                    action: VERSION_RESTORE_ACTION
-                });
-                versions.total_count += 1;
+        if (file_version) {
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: Could you short circuit this by doing 
```
if (!file_version ) {
   return versions; 
}
```

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 4]
```diff
@@ -424,30 +424,45 @@ class ActivityFeed extends React.Component<Props, State> {
...
+            const { modified_by, version_number } = file;
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: if you're pulling the remaining props from file here, maybe only get `file_version` at the top of the method and then get `restored_from`, `modified_at`, `version_number`, etc here?

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 5]
```diff
@@ -424,30 +424,45 @@ class ActivityFeed extends React.Component<Props, State> {
...
+            let currentVersion = file_version;
+            let action = VERSION_UPLOAD_ACTION;
+            let versionNumber = version_number;
+            if (restored_from) {
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: new line before

File: src/components/ContentSidebar/ActivityFeed/activity-feed/__tests__/ActivityFeed-test.js [hunk 6]
```diff
@@ -761,4 +766,30 @@ describe('components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed', ()
             );
         });
     });
+
+    describe('addCurrentVersion()', () => {
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/__tests__/ActivityFeed-test.js]: Not sure if this test already exists but it would be worth adding a test to ensure that the empty state shows up  when only 1 version item is in feedItems and then that the active state shows up when you have 1 version and anything else

File: src/constants.js [hunk 7]
```diff
@@ -227,3 +227,8 @@ export const HTTP_HEAD: 'head' = 'head';
 
 /* ------------------ HTTP Codes  ---------------------- */
 export const UNAUTHORIZED_CODE = 403;
+
+/* ------------------ Version Action Types  ---------------------- */
+export const VERSION_RESTORE_ACTION = 'restore';
```
Comment (pramodsum) [inline src/constants.js]: 👍

File: src/util/fields.js [hunk 8]
```diff
@@ -102,7 +102,13 @@ export const TASKS_FIELDS_TO_FETCH = [
 ];
 
 // Fields needed to get tasks data
-export const VERSIONS_FIELDS_TO_FETCH = [FIELD_TRASHED_AT, FIELD_CREATED_AT, FIELD_MODIFIED_AT, FIELD_MODIFIED_BY];
+export const VERSIONS_FIELDS_TO_FETCH = [
+    FIELD_TRASHED_AT,
+    FIELD_CREATED_AT,
+    FIELD_MODIFIED_AT,
+    FIELD_MODIFIED_BY,
+    FIELD_VERSION_NUMBER
```
Comment (pramodsum) [inline src/util/fields.js]: 👍

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js (hunk 5 above)
Comment (DanDeMicco) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: I think this is a matter of opinion. I would say if you feel strongly about this, introduce an eslint rule, similar to how @jeremypress added one for the blank line after brackets

File: src/api/__tests__/Versions-test.js (hunk 2 above)
Comment (DanDeMicco) [reply src/api/__tests__/Versions-test.js]: 👍 doesn't change the test, but perhaps it would be good to extract the stubs into some constants file so we don't have to keep reimplementing them. Will create a ticket for that

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js (hunk 3 above)
Comment (DanDeMicco) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: 👍

File: src/components/ContentSidebar/ActivityFeed/activity-feed/__tests__/ActivityFeed-test.js (hunk 6 above)
Comment (DanDeMicco) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/__tests__/ActivityFeed-test.js]: I didn't see the test, so added one.

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js (hunk 5 above)
Comment (pramodsum) [reply src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: Sounds good! i'll look into adding one :)

File: src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js [hunk 9]
```diff
@@ -432,40 +432,39 @@ class ActivityFeed extends React.Component<Props, State> {
...
-                action,
-                modified_by,
-                created_at: modified_at,
-                version_number: versionNumber
-            };
-            return {
-                total_count: versions.total_count + 1,
-                entries: [...versions.entries, currentFileVersion]
-            };
+        const { modified_by, version_number } = file;
```
Comment (pramodsum) [inline src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js]: if you're pulling the remaining props from file here, maybe only get `file_version` at the top of the method and then get `restored_from`, `modified_at`, `version_number`, etc here?
